proposals/lip_define_state_and_state_transitions_of_dpos_module.md

```diff
@@ -0,0 +1,1160 @@
+        or currentBFTParameters.certificateThreshold != BFT_THRESHOLD):
+        BFT.setBFTParameters(BFT_THRESHOLD,
+                             BFT_THRESHOLD,
+                             bftWeights)
+
+    if NUMBER_STANDBY_DELEGATES == 2:
+        randomSeed1 = random.getRandomBytes(b.height +1 - (ROUND_LENGTH*3)//2, 
+                                     ROUND_LENGTH)
+        randomSeed2 = random.getRandomBytes(b.height +1 - 2*ROUND_LENGTH,
+                                     ROUND_LENGTH)
+        delegate1, delegate2 = addresses of the standby delegates selected from validatorsTwoRoundsAgo.delegateWeightSnapshot 
+                               as specified in LIP 0022, using the seeds randomSeed1 and randomSeed2
+        # in the above, if validatorsTwoRoundsAgo.delegateWeightSnapshot is empty, then no standby delegates are selected
+        validators = union of activeDelegates and {delegate1, delegate2}
+
+    elif NUMBER_STANDBY_DELEGATES == 1:
+        randomSeed1 = random.getRandomBytes(b.height +1 - (ROUND_LENGTH*3)//2,
+                                 	 ROUND_LENGTH)
+        delegate1 = address of the standby delegates selected from validatorsTwoRoundsAgo.delegateWeightSnapshot 
+                    as specified in LIP 0022, using the seed randomSeed1
+        validators = union of activeDelegates and {delegate1}
+    else: # no standby delegates
+        randomSeed1 = random.getRandomBytes(b.height +1 - (ROUND_LENGTH*3)//2,
+                                 	 ROUND_LENGTH)
+        validators = activeDelegates
+
+    nextValidators = shuffleValidatorsList(validators, randomSeed1)
+    validators.setGeneratorList(nextValidators)
+```
+
+
+### Protocol Logic for Other Modules
+
+More functions might be made available during implementation.
+
+
+#### isNameAvailable
+
+Asserts the availability of a given name for delegate registration.
+
+
+##### Parameters
+
+* `name`: A string being asserted for availability.
+
+
+##### Returns
+
+A boolean asserting the availability of the given name for delegate registration.
+
+
+##### Execution
+
+```python
+isNameAvailable(name):
+    if (nameStore(name) exists
+       or name includes symbols not in "abcdefghijklmnopqrstuvwxyz0123456789!@$&_."
+       or length(name) > MAX_LENGTH_NAME
+       or length(name) < 1):
+        return False
+    else:
+        return True
+```
+
+
+#### getVoter
+
+Returns the stored information relative to the given address.
+
+
+##### Parameters
+
+* `address`:  A 20-byte value identifying the voter.
+
+
+##### Returns
+
+This functions returns `voterStore(address)` deserialized using `voterStoreSchema`.
+
+
+#### getDelegate
+
+Returns the stored information relative to the given address.
+
+
+##### Parameters
+
+* `address`:  A 20-byte value identifying the delegate.
+
+
+##### Returns
+
+This functions returns `delegateStore(address)` deserialized using `delegateStoreSchema`.
+
+
+### Endpoints for Off-Chain Services
+
+
+#### getVoter(address)
```

Comment:
```suggestion
#### getVoter
```

Not sure why we put the parameter in the header here. Same for other off-chain functions

Reply:
removed